baremetal/utils.go

```diff
@@ -44,3 +53,26 @@ type NotFoundError struct {
 func (e *NotFoundError) Error() string {
 	return "Object not found"
 }
+
+func patchIfFound(ctx context.Context, helper *patch.Helper, host runtime.Object) error {
+	err := helper.Patch(ctx, host)
+	if err != nil {
+		notFound := true
+		if aggr, ok := err.(kerrors.Aggregate); ok {
+			for _, kerr := range aggr.Errors() {
+				if !apierrors.IsNotFound(kerr) {
+					notFound = false
```

Comment:
why not returning nil if `apierrors.IsNotFound(kerr)`

Reply:
because it is already deleted, the end goal is achieved

Reply:
not clear, you are anyhow returning nil when `if notFound`

Reply:
oh, I see sorry, I did not get your question right. The problem is that you are getting an aggregate of errors, and I want to return nil if all errors are NotFound. If one of them is not NotFound, I want to return the normal error (or requeue). so we need to check them all, and notFound is true only if all of them are NotFound errors, in which case we return nil